### DISCUSSION TITLE
Pico WiFi is nice, but...

### DISCUSSION THREAD

**Discussion by jazzjohn27** [hathach/tinyusb#1762] [General]
Without an attached keyboard and display (and without running the IDE), there seems to be no easy way for the end user to enter their SSID and PW.

No Bluetooth, so can't use a phone.
Can't read or write a text file that will be available in USB drive mode.
Run an AP that serves a fillable web page? (That's what some smart plugs seem to do.)
Set up an external website that the Pico logs onto for the credentials?

Any other thoughts?

**Comment by hathach** [comment]
I don't understand your question, can you explain more of your hardware setup and user usage/scenario in details.

**Reply by jazzjohn27** [reply]
Raspberry Pi Pico W controls a motor that dispenses a pet treat when the user clicks a button on the webpage served by the Pico W.
In order for this to work for the user, the user's WiFi credentials must be somehow supplied to the Pico W, and the IP assigned to the Pico W must be shared with the user. (For the IP part, there will be a small display to show the user the assigned IP. I am avoiding requiring the user to log onto the router.)

I meant this as a general question about how to get WiFi credentials to the Pico without access to the programming environment and without any "complicated" method for the end user with basic access to their computer, and without any knowledge of using a serial terminal. Also I would like to avoid a keyboard connection to the Pico W.

**Reply by hathach** [reply]
since this is generic picow question, I think you are better ask this on rpi forum. There is much larger users there, also rpi employees who design picow as well.
